public/app/percona/dbaas/components/DBCluster/DBCluster.utils.tsx

```diff
@@ -78,24 +77,48 @@ export const getExpectedResourcesDifference = (
   };
 };
 
-export const formatDBClusterVersion = (version?: string) => (version ? version.split(':')[1].split('-')[0] : '');
+export const formatDBClusterVersion = (version?: string) => version;
 
 export const formatDBClusterVersionWithBuild = (version?: string) => (version ? version.split(':')[1] : '');
 
-const clustersToModel = (database: Databases, clusters: DBClusterPayload[], kubernetes: Kubernetes[], index: number) =>
-  clusters.map((cluster) => {
-    return newDBClusterService(database).toModel(cluster, kubernetes[index].kubernetesClusterName, database);
+const clustersToModel = (clusters: DBClusterListPayload[], kubernetes: Kubernetes[], index: number) =>
+  clusters.map((dbCluster) => {
+    let database = DATABASE_CLUSTER_TYPE[dbCluster.cluster_type] as Databases;
+    return {
+      id: dbCluster.id,
+      clusterName: dbCluster.name,
+      kubernetesClusterName: kubernetes[index].kubernetesClusterName,
+      databaseType: database,
+      installedImage: dbCluster.installed_image,
+      availableImage: dbCluster.available_image,
+    };
   });
 
 export const formatDBClusters = (results: any[], kubernetes: Kubernetes[]) => {
-  const clustersList: DBCluster[] = results.reduce((acc: DBCluster[], r, index) => {
-    const pxcClusters: DBClusterPayload[] = r.pxc_clusters ?? [];
-    const psmdbClusters: DBClusterPayload[] = r.psmdb_clusters ?? [];
-    const pxcClustersModel = clustersToModel(Databases.mysql, pxcClusters, kubernetes, index);
-    const psmdbClustersModel = clustersToModel(Databases.mongodb, psmdbClusters, kubernetes, index);
+  const clustersList: Array<Partial<DBCluster>> = results.reduce((acc: Array<Partial<DBCluster>>, r, index) => {
+    const dbClusters: DBClusterListPayload[] = r.db_clusters ?? [];
+    const dbClustersModel = clustersToModel(dbClusters, kubernetes, index);
 
-    return acc.concat([...pxcClustersModel, ...psmdbClustersModel]);
+    return acc.concat([...dbClustersModel]);
   }, []);
 
   return clustersList;
 };
+
+export const formatDBClusterDetails = (results: any[], db_clusters: Array<Partial<DBCluster>>) => {
+  const clustersList: DBClusterDetails = results.reduce((acc: DBClusterDetails, r, index) => {
+    let cluster = db_clusters[index];
+    let database = cluster.databaseType!;
+
+    let dbCluster = newDBClusterService(database).toModel(
+      cluster,
+      r[database === Databases.mongodb ? 'psmdb_cluster' : 'pxc_cluster'],
```

Comment:
Just a question here, would it be possible to use `CLUSTER_TYPE_DATABASE` here instead of the condition?

Reply:
it's possible, but we need to agree this with BE, because BE named clusters as 'psmdb_cluster' and 'pxc_cluster'

So it takes time and i think it would be better to discuss it with Nurlan. I have next ticket, where I need to move all forms from modal windows to pages and I will do a big refactoring there, so I suppose it should be part of this refactoring

![image](https://user-images.githubusercontent.com/90199600/196692538-934f7e87-26cb-4cea-b5d5-fabc6c7aca51.png)

Reply:
I agree, lets refactor it later.